openstack-mixin/panels.libsonnet

```diff
@@ -0,0 +1,1004 @@
+          },
+        },
+        {
+          id: 'organize',
+          options: {
+            excludeByName: {
+              Time: true,
+              Value: true,
+              __name__: true,
+              hostname: true,
+              job: true,
+            },
+            includeByName: {},
+            indexByName: {
+              Time: 6,
+              Value: 9,
+              __name__: 7,
+              adminState: 3,
+              hostname: 5,
+              id: 4,
+              instance: 1,
+              job: 8,
+              service: 0,
+              zone: 2,
+            },
+            renameByName: {
+              Time: '',
+              adminState: 'Admin state',
+              id: 'ID',
+              instance: 'Instance',
+              service: 'Service',
+              zone: 'Zone',
+            },
+          },
+        },
+      ]),
+
+    volumes:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Volumes',
+        targets=[t.volumes],
+        description='The number of volumes managed by Cinder.',
+      )
+      + timeSeries.standardOptions.withDecimals(0),
+
+    volumeStatus:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Volume status',
+        targets=[t.volumeErrorStatus, t.volumeNonErrorStatus],
+        description='The current status of volumes in Cinder.',
+      )
+      + timeSeries.options.legend.withPlacement('right')
+      + timeSeries.fieldConfig.defaults.custom.withStackingMixin({
+        group: 'A',
+        mode: 'normal',
+      })
+      + timeSeries.standardOptions.withDecimals(0),
+
+    volumeUsage:
+      commonlib.panels.generic.timeSeries.percentage.new(
+        'Volume usage',
+        targets=[t.volumeUsage],
+        description='The percent of volume storage in use for Cinder.',
+      )
+      + timeSeries.standardOptions.withUnit('percentunit')
+      + timeSeries.standardOptions.withMax(1)
+      + timeSeries.standardOptions.withMin(0),
+
+    backupUsage:
+      commonlib.panels.generic.timeSeries.percentage.new(
+        'Backup usage',
+        targets=[t.backupUsage],
+        description='The percent of backup storage in use for Cinder.',
+      )
+      + timeSeries.standardOptions.withUnit('percentunit')
+      + timeSeries.standardOptions.withMax(1)
+      + timeSeries.standardOptions.withMin(0),
+
+    poolUsage:
+      commonlib.panels.generic.timeSeries.percentage.new(
+        'Pool usage',
+        targets=[t.poolUsage],
+        description='The percent of pool capacity in use for Cinder.',
+      )
+      + timeSeries.standardOptions.withUnit('percentunit')
+      + timeSeries.standardOptions.withMax(1)
+      + timeSeries.standardOptions.withMin(0),
+
+    snapshots:
+      commonlib.panels.generic.timeSeries.base.new(
+        'Snapshots',
+        targets=[t.snaphots],
+        description='The number of volume snapshots in Cinder.',
+      )
+      + timeSeries.standardOptions.withDecimals(0),
+
+    cinderAgents:
```

Comment:
For some reason, i don't see any data on this no matter how far back I go.

Reply:
For all tables, the key to having data show up is making sure that there is load present at the very end of your time range. I'm starting my vm back up to make this a little easier.